src/main/java/org/jabref/gui/actions/JabRefAction.java

```diff
@@ -38,11 +38,39 @@ public JabRefAction(Action action, Command command, KeyBindingRepository keyBind
         }
     }
 
+    /**
+     * especially for the track execute when the action run the same function but from different source.
+     * @param source is a string contains the source, for example "button"
+     */
+    public JabRefAction(Action action, Command command, KeyBindingRepository keyBindingRepository, String source) {
```

Comment:
To avoid duplicating code, I suggest modifying the previous method (the _old_ one), for example:
```
public JabRefAction(Action action, Command command, KeyBindingRepository keyBindingRepository) {
    this(action, command, keyBindingRepository, null);
}
```

This way, you should check in your `JabRefAction()` method if `source` is `null` for tracking an execution:
```
setEventHandler(event -> {
    command.execute();

    if (source == null) {
        trackExecute(getActionName(action, command));
    } else {
        trackExecute(String.format("%sFrom%s", getActionName(action, command), source));
    }
});
```

You also can use `Strings.isNullOrEmpty(source)`, from the `com.google.common.base.Strings` package. Or even an `Optional<String>` type.

Reply:
We have our own StringIsNullorEmpty methd in org.jabref.model.StringUtil